src/streamlink/plugins/twitch.py

```diff
@@ -128,6 +135,79 @@
 )
 
 
+Segment = namedtuple("Segment", "uri duration title key discontinuity scte35 byterange date map")
+
+
+class TwitchM3U8Parser(M3U8Parser):
+    def __init__(self, base_uri=None, worker=None, **kwargs):
+        M3U8Parser.__init__(self, base_uri=base_uri, **kwargs)
+        self.worker = worker
+
+    def parse_tag_ext_x_scte35_out(self, value):
+        self.worker.scte35 = True
+
+    def parse_tag_ext_x_scte35_in(self, value):
+        self.worker.scte35 = False
+
+    def get_segment(self, uri):
+        byterange = self.state.pop("byterange", None)
+        extinf = self.state.pop("extinf", (0, None))
+        date = self.state.pop("date", None)
+        map_ = self.state.get("map")
+        key = self.state.get("key")
+        discontinuity = self.state.pop("discontinuity", False)
+
+        return Segment(
+            uri,
+            extinf[0],
+            extinf[1],
+            key,
+            discontinuity,
+            self.worker.scte35,
+            byterange,
+            date,
+            map_
+        )
+
+
+class TwitchHLSStreamWorker(HLSStreamWorker):
+    scte35 = False
+
+    def _reload_playlist(self, text, url):
+        return load_hls_playlist(text, url, parser=TwitchM3U8Parser, worker=self)
+
+
+class TwitchHLSStreamWriter(HLSStreamWriter):
+    def __init__(self, *args, **kwargs):
+        HLSStreamWriter.__init__(self, *args, **kwargs)
+        self.disable_ads = self.reader.stream.options.get("disable-ads") or False
```

Comment:
> self.reader.stream.options.get("disable-ads")

might be already available `self.stream.options.get("disable-ads")`

> or False

you could also forse it **False** in PluginArgument